### PR TITLE
Update Frontend Content to use Wagmi v2

### PR DESCRIPTION
**Pull request by briandoyle81CB**
**What changed? Why?**
Wagmi v2 is very breaking.  Updated tutorials to new methods.

**Notes to reviewers**

**How has it been tested?**

**Does this PR add a new token to the bridge?**

Are you adding an entry to [`assets.ts`](../apps/bridge/assets.ts)?

- [x] No, this PR does not add a new token to the bridge
- [ ] Yes, and I've confirmed this token doesn't use a bridge override
- [ ] Yes, and I've confirmed this token is an OptimismMintableERC20

If you are adding a token to the bridge, please include evidence of both confirmations above for your reviewers.

### CODE REVIEW

File: apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md [hunk 1]
```diff
@@ -63,105 +49,114 @@ Run your app with `yarn dev` to make sure it generated correctly.
...
+You must configure these wrappers in a separate file. It will not work if you try to add them and `"use client":` directly in `layout.tsx`!
+
+:::
+
+Add a new file in the `app` folder called `providers.tsx`.
 
 ### Imports
 
-Start with the imports:
+As discussed above, add ``"use client":` to the top of the file.
```
Comment (taycaldwell) [inline apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md:80]: Extra ` I think on this line?

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/configuring-useReadContract.md [hunk 2]
```diff
@@ -129,21 +137,22 @@ You likely need to share timely updates with your users, but how timely do those
...
-  [baseSepolia],
-  [
-    // other providers...
-    publicProvider(),
-  ],
-  { pollingInterval: 30_000 },
-);
+const config = getDefaultConfig({
+  appName: 'RainbowKit App',
+  projectId: '51407fcf066d74968d9a1a4c6da0d994',
```
Comment (taycaldwell) [inline apps/base-docs/base-camp/docs/reading-and-displaying-data/configuring-useReadContract.md:145]: Should this be a placeholder string?

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/useAccount.md [hunk 3]
```diff
@@ -135,16 +120,16 @@ A common pattern is to use the conditional directly in the html return of a comp
 <div>
   <h2>Connection Information</h2>
   <div>
-    {isConnecting && <p>Please click Connect in your wallet...</p>}
+    {!isConnected && <p>Please click Connect in your wallet...</p>}
```
Comment (taycaldwell) [inline apps/base-docs/base-camp/docs/reading-and-displaying-data/useAccount.md:123]: `isConnecting` is still exists in v2.

**Review by taycaldwell — CHANGES_REQUESTED** [review]
Minor comments. Otherwise LGTM.

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/configuring-useReadContract.md (hunk 2 above)
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/reading-and-displaying-data/configuring-useReadContract.md:145]: Haha whoops

**Pull request merged**